account_invoice_constraint_chronology/model/account_move.py

```diff
@@ -160,8 +169,11 @@ def write(self, vals):
                 move._get_sequence_order_conflicting_invoices_domain(), limit=1
             ):
                 move._raise_sequence_ordering_conflict()
-            if self.search(move._get_older_conflicting_invoices_domain(), limit=1):
-                move._raise_older_conflicting_invoices()
+            older_move = self.search(
+                move._get_older_conflicting_invoices_domain(), limit=1
```

Comment:
Does it make sense to separate the conflict search for entering the condition from the conflict search again for raising the exception? And if it's done in a unified way, wouldn't it be more appropriate to use a constraint instead of putting it in the write method?

Reply:
it's probably better as a constraint... I'm not an expert of this module myself, just got it working for my use case today, I'm not too sure about aggressive refactors. But if you feel you can do it go for it, I'll do my best to review,